src/pages/Access.tsx

```diff
@@ -36,17 +37,40 @@ function extractSwarmHash(string: string): string | null {
   return (matches && matches[0]) || null
 }
 
-function recognizeSwarmHash(value: string) {
-  if (value.length < 64) {
-    return value
+function extractSwarmCid(s: string): string | null {
+  const matches = s.match(/bah[a-z0-9]{58}/)
+
+  if (!matches || !matches[0]) {
+    return null
   }
 
+  const cid = matches[0]
+  try {
+    const decodeResult = decodeCid(cid)
+
+    if (!decodeResult.type) {
```

Comment:
This check will prevent decoding legacy CIDs that used the `dag-pb` codec... Not sure if it is needed or not, just a note.

Reply:
Yeah, I figured, but I think it's okay since this is just a convenience function so that the bzz.links that are generated on the gateway can be copied and accepted as valid references.